### PR TITLE
Revisions from feedback on Smtdeleg/Sstcfg

### PR DESCRIPTION
**Pull request by bcstrongx**

### CODE REVIEW

**Review by pdonahue-ventana — COMMENTED** [review]
Omitting the parenthetical phrases, the introduction says:
> For a Machine-level environment, extension Smtdeleg encompasses all added CSRs and all behavior modifications for a hart, over all privilege levels. For a Supervisor-level environment, extension Sstcfg provides access to delegated triggers. These extensions depend on the Sdtrig and Sscsrind extensions.

That seems like it's missing some previous introductory material.  "Smtdeleg encompasses all added CSRs and all behavior modifications for a hart" for what?

Also, do we need the parenthetical phrases?  They're pretty long and they chop up the text and make it harder to read.  For instance, I don't see similar justifications for other recent extensions in the priv spec.

**Comment by bcstrongx** [comment]
@pdonahue-ventana honestly I just lifted that from the Smcdeleg/Ssccfg spec, and in that case I think the test was requested by ARC.

https://github.com/riscv/riscv-smcdeleg-ssccfg/blob/main/intro.adoc

But that's lazy, and it did get nixed from the priv spec port.  Between your feedback and Tim's, I think a better introduction is in order.  WIP.

**Comment by bcstrongx** [comment]
Introduction is revised

**Review by pdonahue-ventana — APPROVED** [review]
I like the new intro.  And the other changes also look good.

File: riscv-smtdeleg-sstcfg.adoc [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
+Sdtrig triggers allow both self-hosted and external debuggers to debug target software by transferring control to the debugger at points of interest.  However, Sdtrig CSRs are accessible only with M-mode privilege.  As a result, a debugger must have M-mode privilege, or be able to invoke M-mode, in order to access them.  
+
+For a self-hosted debugger with less than M-mode privilege, this requires an SBI (Supervisor Binary Interface).  This results in traps on all trigger accesses, which can add latency to critical context switch flows.
+
+For an external debugger, if Sdsec is implemented and the hart is configured such that Debug Mode has less than M-mode privilege, the debugger is unable to access triggers.  There is no SBI that can be called from Debug Mode.
+
+The Smtdeleg extension provides a means for Sdtrig triggers to be accessed from S-mode and VS-mode, using the indirect CSR interface (Sscsrind).  It also defines a means to control S-mode and VS-mode access to triggers using the State Enable extensions (Smstateen/Ssstateen).  Sstcfg comprises the supervisor components of Smtdeleg.
```
Comment (rtwfroody) [inline riscv-smtdeleg-sstcfg.adoc:10]: Very clear. Thanks.

**Review by joxie — APPROVED** [review]
LTGM

**Pull request merged**